test/new-e2e/examples/vm_test.go

```diff
@@ -0,0 +1,164 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package examples
+
+import (
+	"flag"
+	"io/fs"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+)
+
+var (
+	devMode = flag.Bool("devmode", false, "enable dev mode")
+)
+
+type vmSuite struct {
+	e2e.Suite[e2e.VMEnv]
+}
+
+// TestVMSuite runs tests for the VM interface to ensure its implementation is correct.
+func TestVMSuite(t *testing.T) {
+	suiteParams := make([]func(*params.Params), 0)
+	if *devMode {
+		suiteParams = append(suiteParams, params.WithDevMode())
+	}
+	e2e.Run(t, &vmSuite{},
+		e2e.EC2VMStackDef(ec2params.WithOS(ec2os.WindowsOS)),
+		suiteParams...,
+	)
+}
+
+func (v *vmSuite) TestExecute() {
+	vm := v.Env().VM
+
+	out, err := vm.ExecuteWithError("whoami")
+	v.Require().NoError(err)
+	v.Require().NotEmpty(out)
+}
+
+func (v *vmSuite) TestFileOperations() {
```

Comment:
I am not testing `sftp` package is actually necessary.

Reply:
`sftp` is currently the only implementation of the `VM` interface. I thought it could be useful to have some basic tests of that interface to ensure that other implementations do the right thing. Plus I needed some way to ensure the wrappers in `vm_client.go` and `ssh.go` were correct.

If there's never going to be another interface and we're always going to use `sftp` it'd be more convenient to expose the `ssh.Client` instead of wrapping everything.